tuxemon/core/event/actions/transition_teleport.py

```diff
@@ -57,15 +57,19 @@ class TransitionTeleportAction(EventAction):
     ]
 
     def start(self):
-        # Start the screen transition
-        params = [self.parameters.transition_time]
-        self.transition = self.game.event_engine.get_action("screen_transition", params)
-        self.transition.start()
+        # Start the screen transition if one isn't already underway
+        world = self.game.get_state_name("WorldState")
+        if not world.in_transition:
+            params = [self.parameters.transition_time]
+            self.transition = self.game.event_engine.get_action("screen_transition", params)
+            self.transition.start()
 
     def update(self):
-        if not self.transition.done:
+        if not hasattr(self, 'transition'):
+            self.stop()
```

Comment:
if we never get a transition, what will cause this action to stop?

Reply:
I had to the hasattr check after noticing that not setting a transition in `def start(self)` would cause this update statement to crash. I could just use a pass statement for it... `self.stop()` was just me wanting to be extra cautious and ensure everything was stopped accordingly.

Reply:
I just tested by replacing `self.stop()` with `pass`. Apparently that stop is needed: Without it, teleport events stop working after two teleports. No errors or warnings, but after being teleported twice on the same map touching the event stops doing anything.